src/Common/ChartVersionAndTypeSelector.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useState } from 'react'
+import { fetchSelectedChartTemplate } from './Common.service'
+import { ChartVersionAndTypeSelectorProps, DeploymentChartVersionType } from './Types'
+import ReactSelect from 'react-select'
+import { customStyles, getFilteredChartVersions, showError } from './Helper'
+
+const ChartVersionAndTypeSelector = ({ setSelectedChartRefId }: ChartVersionAndTypeSelectorProps) => {
+    const [charts, setCharts] = useState<DeploymentChartVersionType[]>([])
+    const [selectedChartType, setSelectedChartType] = useState(null)
+    const [chartVersionOptions, setChartVersionOptions] = useState([])
+    const [chartTypeOptions, setChartTypeOptions] = useState([])
+    const [selectedChartVersion, setSelectedChartVersion] = useState(null)
+
+    useEffect(() => {
+        fetchSelectedChartTemplate()
+            .then((res) => {
+                setCharts(res?.result)
```

Comment:
We should add a check for res?.result rather than checking it on multiple places